argon2-jvm-nolibs/src/main/java/de/mkammerer/argon2/BaseArgon2.java

```diff
@@ -362,4 +419,84 @@ private boolean verifyBytes(String hash, byte[] pwd) {
         Arrays.fill(byteBuffer.array(), (byte) 0); // clear sensitive data
         return bytes;
     }
+
+    /**
+     * Builds a {@link Argon2_context} by the specified arguments.
+     *
+     * @param iterations     Iterations.
+     * @param memory         Memory.
+     * @param parallelism    Parallelism.
+     * @param hashLength     Hash length.
+     * @param password       Password.
+     * @param salt           Salt.
+     * @param secret         Secret (nullable).
+     * @param associatedData Associated Data (nullable).
+     * @param version        Version (nullable).
+     * @return {@link Argon2_context}
+     */
+    private static Argon2_context.ByReference buildContextReference(int iterations, int memory, int parallelism, int hashLength, byte[] password, byte[] salt, byte[] secret, byte[] associatedData, Argon2Version version) {
+        Argon2_context.ByReference context = new Argon2_context.ByReference();
+
+        context.out = new Memory(hashLength);
+        context.outlen = new JnaUint32(hashLength);
+
+        context.pwd = new Memory(password.length);
+        context.pwd.write(0, password, 0, password.length);
+        context.pwdlen = new JnaUint32(password.length);
+
+        context.salt = new Memory(salt.length);
+        context.salt.write(0, salt, 0, salt.length);
+        context.saltlen = new JnaUint32(salt.length);
+
+        if (secret != null) {
+            context.secret = new Memory(secret.length);
+            context.secret.write(0, secret, 0, secret.length);
+            context.secretlen = new JnaUint32(secret.length);
+        } else {
+            context.secret = Pointer.NULL;
+            context.secretlen = new JnaUint32(0);
+        }
+
+        if (associatedData != null) {
+            context.ad = new Memory(associatedData.length);
+            context.ad.write(0, associatedData, 0, associatedData.length);
+            context.adlen = new JnaUint32(associatedData.length);
+        } else {
+            context.ad = Pointer.NULL;
+            context.adlen = new JnaUint32(0);
+        }
+
+        context.t_cost = new JnaUint32(iterations);
+        context.m_cost = new JnaUint32(memory);
+        context.lanes = new JnaUint32(parallelism);
```

Comment:
`lanes` and `threads` are set to the same value (`parallelism`), did you find any documentation on that?

Reply:
I tried to make these methods as similar as possible to the C library.

They used these lines while creating their `context struct` for the base `argon2_hash` function:
```C
context.lanes = parallelism;
context.threads = parallelism;
```
You can find the function [here](https://github.com/P-H-C/phc-winner-argon2/blob/16d3df698db2486dde480b09a732bf9bf48599f9/src/argon2.c#L100).

Reply:
Great, please just add a comment to that file. this will save me some searching if i stumble over that again :) Thanks!

Reply:
Sure, no problem.